tests/unit/expiry/ExpiryDateStrategyFactory.unitTest.ts

```diff
@@ -40,71 +47,59 @@ describe("ExpiryDateStrategyFactory", () => {
             invalidConfig
           );
         } catch (error) {
-          expect.assertions(1);
           expect(error).toEqual(new Error(ERRORS.ExpiryConfigMissing));
         }
       });
     });
 
     describe("when valid test type id is passed and more than one strategy is returned", () => {
-      it("should throw error", () => {
+      it("should throw an appropriate error", () => {
+        expect.assertions(1);
         try {
+          //  @ts-ignore
+          jest.spyOn(ExpiryDateStrategyFactory, 'getStrategyMapping').mockReturnValue(duplicatedConfig.psv);
+
           const testTypeForExpiry: TestTypeForExpiry = {
             testType: { testTypeId: "3" } as TestType,
             vehicleType: VEHICLE_TYPE.PSV,
             recentExpiry: new Date(0),
             hasHistory: true,
-            hasRegistration: false
+            hasRegistration: true
           };
           ExpiryDateStrategyFactory.GetExpiryStrategy(
             testTypeForExpiry,
             new DateProvider()
           );
         } catch (error) {
-          expect.assertions(1);
           expect(error).toEqual(new Error("Multiple strategies found!"));
         }
       });
     });
-  });
-  context("for valid strategy mapping", () => {
-    describe("when wrong test type id is passed", () => {
-      it("should return error on getExpiryDate", () => {
-        try {
-          const testTypeForExpiry: TestTypeForExpiry = {
-            testType: { testTypeId: "100" } as TestType,
-            vehicleType: VEHICLE_TYPE.HGV,
-            recentExpiry: new Date(0),
-            hasHistory: true,
-            hasRegistration: false
-          };
-          ExpiryDateStrategyFactory.GetExpiryStrategy(
-            testTypeForExpiry,
-            new DateProvider()
-          ).getExpiryDate();
-        } catch (error) {
-          expect.assertions(1);
-          expect(error).toEqual(new Error(ERRORS.MethodNotImplemented));
-        }
-      });
-    });
 
-    describe("when correct test type id is passed", () => {
-      it("should return correct strategy", () => {
+    describe("Resolving correct strategies", () => {
+      test.each`
+      hasHistory  |  hasRegistration  | testTypeId  | vehicleType         | expectedStrategy
+      ${true}     |  ${false}         | ${'3'}      | ${VEHICLE_TYPE.PSV} | ${PsvMostRecentExpiryStrategy}
+      ${true}     |  ${true}          | ${'3'}      | ${VEHICLE_TYPE.PSV} | ${PsvMostRecentExpiryStrategy}
+      ${false}    |  ${true}          | ${'3'}      | ${VEHICLE_TYPE.PSV} | ${PsvRegistrationAnniversaryStrategy}
+      ${false}    |  ${false}         | ${'3'}      | ${VEHICLE_TYPE.PSV} | ${PsvDefaultExpiryStrategy}
+      ${true}     |  ${false}         | ${'100'}    | ${VEHICLE_TYPE.HGV} | ${NoImplementationStrategy}
```

Comment:
This is really good.  Can we add the following table entries to cover the other mapping scenario's then we are checking them all going forward?

```suggestion
      ${true}     |  ${false}         | ${'100'}    | ${VEHICLE_TYPE.HGV} | ${NoImplementationStrategy}
      ${true}     |  ${false}         | ${'100'}    | ${VEHICLE_TYPE.HGV} | ${NoImplementationStrategy}
      ${true}     |  ${true}          | ${'146'}    | ${VEHICLE_TYPE.PSV} | ${PsvDefaultExpiryStrategy}
      ${true}     |  ${false}         | ${'146'}    | ${VEHICLE_TYPE.PSV} | ${PsvDefaultExpiryStrategy}
      ${false}    |  ${false}         | ${'146'}    | ${VEHICLE_TYPE.PSV} | ${PsvDefaultExpiryStrategy}
      ${false}    |  ${true}          | ${'64'}     | ${VEHICLE_TYPE.HGV} | ${HgvTrlFirstTestStrategy}
      ${false}    |  ${false}         | ${'64'}     | ${VEHICLE_TYPE.HGV} | ${HgvTrlFirstTestStrategy}
      ${false}    |  ${true}          | ${'54'}     | ${VEHICLE_TYPE.HGV} | ${HgvTrlAnnualTestStrategy}
      ${false}    |  ${false}         | ${'54'}     | ${VEHICLE_TYPE.HGV} | ${HgvTrlAnnualTestStrategy}
      ${true}     |  ${true}          | ${'54'}     | ${VEHICLE_TYPE.HGV} | ${HgvTrlMostRecentExpiryStrategy}
      ${true}     |  ${false}         | ${'54'}     | ${VEHICLE_TYPE.HGV} | ${HgvTrlMostRecentExpiryStrategy}
      ${false}    |  ${false}         | ${'65'}     | ${VEHICLE_TYPE.TRL} | ${HgvTrlFirstTestStrategy}
      ${false}    |  ${true}          | ${'65'}     | ${VEHICLE_TYPE.TRL} | ${HgvTrlFirstTestStrategy}
      ${false}    |  ${false}         | ${'54'}     | ${VEHICLE_TYPE.TRL} | ${HgvTrlAnnualTestStrategy}
      ${false}    |  ${true}          | ${'54'}     | ${VEHICLE_TYPE.TRL} | ${HgvTrlAnnualTestStrategy}
      ${true}     |  ${true}          | ${'65'}     | ${VEHICLE_TYPE.TRL} | ${HgvTrlMostRecentExpiryStrategy}
      ${true}     |  ${false}         | ${'65'}     | ${VEHICLE_TYPE.TRL} | ${HgvTrlMostRecentExpiryStrategy}
```

Reply:
Done